src/iop/lens.cc

```diff
@@ -367,6 +367,11 @@ static lfModifier * get_modifier(int *mods_done, int w, int h, const dt_iop_lens
   return mod;
 }
 
+static inline gboolean monochrome_img(const dt_image_t *img)
```

Comment:
Shouldn't this be `dt_image_is_monochrome()` in image.c and fixed there as the current implementation does not seems to take bayer image:

```
int dt_image_is_monochrome(const dt_image_t *img)
{
  return (img->flags & DT_IMAGE_MONOCHROME);
}
```

Reply:
BTW, looking at the current usage we can safely return a gboolean here.

Reply:
Fully agreed - principally. Yet - there are a few cornercases using the current implementation of `dt_image_is_monochrome` over several sources that would need more changes. There is a larger pr following soon for 4.0 that fixes the whole stuff. There are even more functions that would be better defined as returning gbooleans. So - i would prefer to keep this pr "as-is"

Reply:
Understood and works for me. I'll review again with this in mind. Thanks.